utils/downloads.go

```diff
@@ -0,0 +1,91 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/cheggaaa/pb.v1"
+)
+
+var snapshots = map[string]string{
+	"mainnet": "https://juno-snapshot.s3.us-east-2.amazonaws.com/mainnet/juno_mainnet_v0.5.0_136902.tar",
+	"goerli":  "https://juno-snapshot.s3.us-east-2.amazonaws.com/goerli/juno_goerli_v0.5.0_839969.tar",
+	"goerli2": "https://juno-snapshot.s3.us-east-2.amazonaws.com/goerli2/juno_goerli2_v0.5.0_135973.tar",
+}
+
+const (
+	downloadBufferSize = 8192 // 8KB
+	dirPermissionMod   = 0o770
+)
+
+type Downloader interface {
```

Comment:
I haven't delved in 100% yet so I could be wrong, but I don't think an interface is necessary here.

I think we can have a signature like this
```go
func(context.Context, url, dir string) error
```
To test, we use `httptest`+`testing.T.TempDir`. This gives us an end-to-end black-box test that runs quickly.

Reply:
I'm not sure how can we mock different download scenarios without the interface.